### ISSUE TITLE
Live streaming

### ISSUE THREAD

**Issue by matthijsotterloo** [mgp25/Instagram-API#1475]
Hi guys!

This question is especially for @solomax2014, I tried streaming to the RTMPS URL Instagram returns but I think we're still missing out on something as I'm unable to open a stream to the URL (it does generate the stream and start it, so it does show up in the app) but as soon as I stream video to that RTMPS URL I don't see it in the Instagram app (I tried this with multiple apps), I think we're missing a key or something. Can anyone confirm this?

Thanks!

**Comment by solomax2014** [comment]
@matthijsotterloo, at first you need to use ```$ig->live->create($height, $width, $broadcastMessage = "")``` to get a link and broadcast ID. Then you can stream to this url, but live will be started in app only after you call ```$ig->live->start($broadcastID)```.

**Comment by matthijsotterloo** [comment]
@solomax2014 Yes, I ran those 2 commands and then opened a stream to the URL from the first response but for some reason it fails to connect with all libs/tools I tried.

**Comment by solomax2014** [comment]
@matthijsotterloo, I can tell you some ways to stream that I know.
- ffmpeg on Windows
Stream from video: [#1446 (comment)](https://github.com/mgp25/Instagram-API/issues/1446#issuecomment-320125785)
Stream from webcam and microphone: [#1446 (comment)](https://github.com/mgp25/Instagram-API/issues/1446#issuecomment-320134960)
Notice: as webcam can use different resolutions and framerates, at first you need to get a list of supported modes (example command for outputs in comments that I mentioned):
```
ffmpeg -f dshow -list_options true -i video="Logitech HD Webcam C270":audio="Microphone (HD Webcam C270)"
```
And then select the resolution and framerate (example command, -vf "transpose=1" to rotate the image by 90 degrees clockwise):
```
ffmpeg -video_size 1280x720 -framerate 30 -rtbufsize 256M -re -f dshow -i video="Logitech HD Webcam C270":audio="Microphone (HD Webcam C270)" -vf "transpose=1" -acodec libmp3lame -ar 44100 -b:a 128k -pix_fmt yuv420p -profile:v baseline -s 720x1280 -bufsize 6000k -vb 400k -maxrate 1500k -deinterlace -vcodec libx264 -preset veryfast -g 30 -r 30 -f flv "upload_url"
```
- OBS Studio on Windows or Mac
You need a little bit to change the url: [#1446 (comment)](https://github.com/mgp25/Instagram-API/issues/1446#issuecomment-321038570)

All these methods are tested by me and @rudemex.

**Comment by cengizakcan1996** [comment]
Hello guys,

can who help me?

i want to rtmp to rtmp 
is this a norm query mp4 to rtmp but i dont want i want rtmp to rtmp

**Comment by solomax2014** [comment]
@cengizakcan1996, it wasn't tested. But if you want to test, try to use ffmpeg, Find the optimal parameters to resize the source stream and use ```-i "<RTMP URL>"``` to use the RTMP stream as a source.

**Comment by cengizakcan1996** [comment]
@solomax2014 can you write example for me ?

**Comment by solomax2014** [comment]
@cengizakcan1996, 
```ffmpeg -rtbufsize 256M -re -i "<source RTMP url>" -acodec libmp3lame -ar 44100 -b:a 128k -pix_fmt yuv420p -profile:v baseline -s 720x1280 -bufsize 6000k -vb 400k -maxrate 1500k -deinterlace -vcodec libx264 -preset veryfast -g 30 -r 30 -f flv "<destination RTMP url>"```
This should work on Windows, I think. I'm not sure about other OS.

**Comment by cengizakcan1996** [comment]
@solomax2014 Thanx :) 👍

**Comment by AnitaPazhohesh** [comment]
Hi,
Please help me how to broadcast live to Instagram from desktop using obs or wircast. I have senn some tv channels doing live on Insta not using a mobile phone! How can I obtain stream key?

**Comment by solomax2014** [comment]
@AnitaPazhohesh, do you know how to use this library in PHP?

**Comment by AnitaPazhohesh** [comment]
Hi,
No! Can you please explain in simpler way how  can I get the stream key for Instagram live from desktop? Is that possible at all?
Thanks for your time and response.
Anita

**Comment by alwayzgame** [comment]
i dont know how to use this library struggling can u please help me

**Comment by solomax2014** [comment]
@alwayzgame, please, read this:
- https://github.com/mgp25/Instagram-API/blob/master/README.md
- https://github.com/mgp25/Instagram-API/tree/master/examples
- About streaming: https://github.com/mgp25/Instagram-API/issues/1475#issuecomment-321578664

**Comment by lemkova** [comment]
is this stuff still works?

last time i tried this, it wont connect to stream server.

**Comment by Artclock** [comment]
Hi 
How About Stream key for OBS?
OBS  need stream key to connect to Instagram for live broadcasting!!

**Comment by mgp25** [comment]
@Artclock please, take a moment and read everything before posting anything. It is well explained.

**Comment by Artclock** [comment]
@mgp25 sure , but for last 3 days i was reading all that.
Just cant find anything about "stream name or key" field in OBS.!

**Comment by demo7up** [comment]
I'm at work right now so I can't try this but thinking through this issue. I've read in multiple places that the new Instagram App for Windows 10 supports USB Cameras. So I was wondering if anyone has tried to use -> OBS Virtual Cam with it ??? https://obsproject.com/forum/resources/obs-virtualcam.539/

If that works it could be a cool solution till we get RTMP access?!?! Just Curious!

**Comment by Daricly** [comment]
Hello guys！
I want to create a live broadcasting on instagram 
but it doesn't worked?

**Comment by penteseter123** [comment]
Please help the master here i want to stream using vmix software but i have not found the key stream whether it can be aided thank you and how

**Comment by arshikh** [comment]
Is it possible to livestream a pre recorded video in Instagram? Is there any way to do that? I don't know whether it's the right place to ask. Reply from any expert would be helpful.

**Comment by 10S10** [comment]
Is there any documentation with live broadcasting API for Instagram on web?

We want to use it for nodejs, so I can try to develop the library for the npm.
